playbooks/generate-csrs.yml

```diff
@@ -0,0 +1,95 @@
+---
+# Generate new certificate signing requests and keys for all cluster certificates,
+# to be delivered to a CA to sign. Keys are left in the csr staging directory
+# on each host, and csr files are copied to localhost in playbooks/csr.tgz
+
+- import_playbook: init/main.yml
+
+# Create CSRs
+- name: Create master CSRs
+  hosts: oo_masters_to_config
+  tasks:
+  - import_role:
+      name: openshift_master_certificates
+      tasks_from: csr.yml
+
+- name: Create etcd CSRs
+  hosts: oo_etcd_to_config
+  tasks:
+  - import_role:
+      name: etcd
+      tasks_from: csr.yml
+
+- name: Create node CSRs
+  hosts: oo_nodes_to_config
+  tasks:
+  - import_role:
+      name: openshift_node_certificates
+      tasks_from: csr.yml
+
+- name: Create hosted CSRs
+  hosts: oo_first_master
+  tasks:
+  - import_role:
+      name: openshift_hosted
+      tasks_from: csr.yml
+    vars:
+      openshift_ca_host: "{{ groups.oo_first_master.0 }}"
+      openshift_csr_dir: "{{ openshift.common.config_base }}/csr"
+    when:
+    - openshift_hosted_manage_registry | default(False) | bool
+
+# Fetch all CSRs.
+- name: Create a local CSR temp dir
+  hosts: oo_first_master
+  tasks:
+  - local_action: command mktemp -d /tmp/openshift-ansible-XXXXXXX
```

Comment:
why not use the tempfile module ?

Reply:
perhaps with a conditional to run only if the directory hasn't been created yet ?
also why /tmp ?
why not a directory in /run or /var somewhere that cannot be raced by regular users and therefore can use a normal directory without using mktemp ?

Reply:
No specific reason. I believe using local_action here is the same as tempfile delegated to localhost. This dir is just for fetching the csrs locally into to create the tarball a run and then its deleted. The pending csrs and keys are still held on each node. I can relocate it as suggested, but I'm not sure that we would really need to worry about it.